dask/array/tests/test_array_core.py

```diff
@@ -873,12 +873,29 @@ def test_blockdims_from_blockshape():
 
 
 def test_coerce():
-    d = da.from_array(np.array([1]), chunks=(1,))
+    d0 = da.from_array(np.array(1), chunks=(1,))
+    d1 = da.from_array(np.array([1]), chunks=(1,))
     with dask.set_options(get=dask.get):
-        assert bool(d)
-        assert int(d)
-        assert float(d)
-        assert complex(d)
+        for d in d0, d1:
+            assert bool(d) is True
+            assert int(d) == 1
+            assert float(d) == 1.0
+            assert complex(d) == complex(1)
+
+    a2 = np.arange(2)
+    d2 = da.from_array(a2, chunks=(2,))
+    with pytest.raises(TypeError):
+        int(d2)
+        float(d2)
+        complex(d2)
```

Comment:
This might not check the latter functions.   I recommend something like the following:

```python
for func in [int, float, complex]:
    with pytest.raises(TypeError):
        func(d2)
```